src/Magnum/DartIntegration/DartSkeleton.h

```diff
@@ -0,0 +1,140 @@
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::DartIntegration::DartSkeleton
+ */
+
+#include <vector>
+
+#include <dart/dynamics/BodyNode.hpp>
+#include <dart/dynamics/ShapeNode.hpp>
+#include <dart/dynamics/Skeleton.hpp>
+
+#include <Magnum/DartIntegration/DartObject.h>
+
+namespace Magnum { namespace DartIntegration {
+/**
+@brief Dart Physics Skeleton (i.e., Multi-Body entity)
+
+Parses a Dart `Skeleton` for easy usage in Magnum. It basically
+parses the `Skeleton` and keeps track of a list of @ref DartObject
+
+## Usage
+
+Common usage is to create a Dart `SkeletonPtr` and then instantiate 
+a `DartSkeleton` by passing a parent @ref SceneGraph feature and 
+the `SkeletonPtr` to its constructor:
+
+@code
+dart::dynamics::SkeletonPtr skel = createSkeletonInDart();
+SceneGraph::Object<SceneGraph::MatrixTransformation3D> object;
+DartSkeleton* dartSkel = new DartSkeleton{&object, skel};
+
+// only the Dart Skeleton can affect the transformation of the Magnum
+// DartSkeleton and not the other way around
+// to get the latest Dart transformations, you should update the objects
+// of dartSkel
+// dartSkel->updateObjects();
+@endcode
+
+## Limitations
+
+- `SoftBodyNode`s are not supported yet
+- When changing the structure of a `SkeletonPtr`, @ref DartObject will not automatically update
+*/
+
+class MAGNUM_DARTINTEGRATION_EXPORT DartSkeleton {
+    public:
+         /**
+         * @brief Constructor
+         * @param parent    parent Object
+         * @param skeleton  Dart SkeletonPtr to parse
+         */
+        template <typename T> explicit DartSkeleton(T* parent = nullptr, dart::dynamics::SkeletonPtr skeleton = nullptr) {
+            if (skeleton && parent)
+                parseSkeleton(parent, skeleton);
+        }
+
+        /**
+         * @brief parse a Dart Skeleton
+         * @param parent    parent Object
+         * @param skeleton  Dart SkeletonPtr to parse
+         */
+        template <typename T> DartSkeleton& parseSkeleton(T* parent, dart::dynamics::SkeletonPtr skeleton) {
+            // @todo Support more than one kinematic trees
+            if (parent && skeleton) {
+                _objects.clear();
+                _parseBodyNodeRecursive(parent, skeleton->getRootBodyNode());
+            }
+
+            return *this;
+        }
+
+        /** @brief update all child objects */
+        DartSkeleton& updateObjects();
+
+        /** @brief get all objects */
+        std::vector<DartObject*> objects();
+        /** @brief get all objects associated with `ShapeNode`s */
+        std::vector<DartObject*> shapeObjects();
+        /** @brief get all objects associated with `BodyNode`s */
+        std::vector<DartObject*> bodyObjects();
+
+    protected:
+        std::vector<DartObject*> _objects;
+
+        template <typename T> void _parseBodyNodeRecursive(T* parent, dart::dynamics::BodyNode* bn) {
+            if (!bn || !parent)
+                return;
```

Comment:
As far as I can see, `nullptr` is never passed to `parent`. Could that be made into a reference instead? Is the same possible for the `bn` parameter? Would make the code a bit clearer.